SerialPrograms/Source/PokemonLA/Programs/PokemonLA_LeapPokemonActions.cpp

```diff
@@ -195,21 +196,27 @@ bool check_tree_or_ore_for_battle(ConsoleHandle& console, BotBaseContext& contex
     MountDetector mount_detector;
 
     QImage snapshot = console.video().snapshot();
-
-
-    for (int i = 0; i < 2; i++) {
-        MountState mount = mount_detector.detect(snapshot);
-
-        if (mount != MountState::NOTHING){
-           console.log("Battle not found. Tree or ore might be empty.");
-           return false;
+    MountState mount = mount_detector.detect(snapshot);
+    bool found = false;
+    for (int i = 0; i < 4; i++) {
+        mount = mount_detector.detect(snapshot);
+
+        if (mount == MountState::NOTHING){
+            found = true;
+            break;
         }
         else{
-            pbf_wait(context, 1 * TICKS_PER_SECOND);
+            console.log("Attempt " + std::to_string(i) + "Found: " + MOUNT_STATE_STRINGS[(int)mount]);
+            pbf_wait(context, 0.5 * TICKS_PER_SECOND);
```

Comment:
Do you need to call `context.wait_for_all_requests();` here to wait for the microcontroller to finish executing commands?

Reply:
yeah forgot to add that! :D